packages/flutter/test/widgets/listener_deprecated_test.dart

```diff
@@ -0,0 +1,623 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter/gestures.dart';
+
+
+// The tests in this file are moved from listener_test.dart, which tests several
+// deprecated APIs. The file should be removed once these parameters are.
+
+// ignore_for_file: deprecated_member_use_from_same_package
+// We have to ignore the lint rule here because we need to use the deprecated
+// callbacks in order to test them.
+
+class HoverClient extends StatefulWidget {
+  const HoverClient({Key key, this.onHover, this.child}) : super(key: key);
+
+  final ValueChanged<bool> onHover;
+  final Widget child;
+
+  @override
+  HoverClientState createState() => HoverClientState();
+}
+
+class HoverClientState extends State<HoverClient> {
+  static int numEntries = 0;
```

Comment:
Can this be non static? if you want to query the numEntries during the test you can use tester.state<HoverClientState>(find.byType(HoverClient))
and then you don't need to worry about previously test effect the current test.

Reply:
This file is entirely moved from `listener_test.dart` and only serves as compatibility check. I'll not change this file. I'll take a look at the one in `mouse_region_test.dart`.